src/components/Select/NativeSelect.vue

```diff
@@ -0,0 +1,176 @@
+<template>
+  <div class="native-select" :class="classes.select">
+    <div class="native-select__input-container">
+      <select ref="input" v-model="selected" class="native-select__input" v-bind="{ disabled, required }">
+        <template v-for="(optionOrGroup, i) in options" :key="i">
+          <template v-if="isOptionGroup(optionOrGroup)">
+            <optgroup :label="optionOrGroup.label" :disabled="optionOrGroup.disabled">
+              <template v-for="(option, x) in optionOrGroup.options" :key="x">
+                <option :value="option.value" :disabled="option.disabled">{{ option.label }}</option>
+              </template>
+            </optgroup>
+          </template>
+          <template v-else>
+            <option :value="optionOrGroup.value" :disabled="optionOrGroup.disabled">{{ optionOrGroup.label }}</option>
+          </template>
+        </template>
+      </select>
+      <div class="native-select__overlay">
+        <template v-if="isSelected">
+          <OptionLabel v-bind="selectedOption">
+            <template v-slot:label="scope">
+              <slot name="selected-option-label" v-bind="scope" />
+            </template>
+            <template v-slot:icon="scope">
+              <slot name="selected-option-icon" v-bind="scope" />
+            </template>
+          </OptionLabel>
+        </template>
+        <template v-else>
+          <span class="native-select__placeholder">
+            {{ placeholder }}
+          </span>
+        </template>
+        <i class="native-select__arrow pi pi-arrow-down-s-line pi-lg" />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, prop } from 'vue-class-component'
+import { Options } from './types'
+import { getOptionFromOptionsAndGroupsByValue, isOptionGroup } from './utilities'
+import { Component } from '@/utilities/vue-class-component'
+import OptionLabel from './OptionLabel.vue'
+
+class Props {
+  modelValue = prop<string | null>({ default: null })
+  options = prop<Options>({ default: () => ({}) })
+  placeholder = prop<string>({ default: 'Choose an Option' })
+  disabled = prop<boolean | string>({ default: false })
+  required = prop<boolean | string>({ default: false })
+}
+
+@Component({
+  emits: ['update:modelValue'],
+  components: {
+    OptionLabel
+  }
+})
+export default class NativeSelect extends Vue.with(Props) {
+
+  $refs!: {
+    input: HTMLSelectElement
+  }
+
+  private isOptionGroup = isOptionGroup
```

Comment:
could have put this in your setup function and avoided the weird declaration right?

Reply:
class components are transpiled to setup. I don't think there's another way to do this in a class component.